### PR TITLE
Test/single event page

### PR DESCRIPTION
**Pull request by yiranli624**
#### What's this PR do?
Add e2e tests for event view
refractor existing tests - replace intercept with request

#### What does it fix?
n/a

#### Where should the reviewer start?
EventView_test.js Line 1;
Postings_test.js Line 7;
User_test.js Line 6;

#### How should this be manually tested?
run ```npm run cypress```

#### Related issues

### CODE REVIEW

File: cypress/integration/EventView_test.js [hunk 1]
```diff
@@ -64,5 +64,50 @@ describe("Testing the single event details page", () => {
...
+          id: "posting-1",
+          eventId: "event-1",
+          eventName: "Something Crazy",
+          positionName: "assisting with check-in",
+          date: "2021/02/01"
+        },
+      }
+    );
+  })
+
```
Comment (okalimbet) [inline cypress/integration/EventView_test.js:83]: need to fix whitespace

File: cypress/integration/EventView_test.js [hunk 2]
```diff
@@ -64,5 +64,50 @@ describe("Testing the single event details page", () => {
     cy.url().should("include", "/")
   })
 
+  it.only("Should display upcoming jobs and able to sign up", () => {
+    cy.get(".posting-position-cards-wrapper")
+      .should("contain", "assisting with check-in")
+      .and("contain", "Open Spots:")
+      .and("contain", "3")
+      .and("contain", "tossing color")
+      .and("contain", "5")
```
Comment (okalimbet) [inline cypress/integration/EventView_test.js]: nice chain!

File: cypress/integration/EventView_test.js [hunk 3]
```diff
@@ -99,14 +105,18 @@ describe("Testing the single event details page", () => {
...
           eventId: "event-1",
           eventName: "Something Crazy",
           positionName: "assisting with check-in",
           date: "2021/02/01"
         },
       }
     );
+    
+    cy.get(".submit-button").should("be.disabled")
+    // cy.get(".event-job-title:first")
```
Comment (okalimbet) [inline cypress/integration/EventView_test.js]: Is this for the future tests or should we delete it?

File: cypress/integration/EventView_test.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+describe("Testing the single event details page", () => {
+
+  beforeEach(() => {
+    cy.visit("http://localhost:3000/postings/event-1");
+    cy.request("GET", "https://ivolunteer-api-test.herokuapp.com/events/event-1");
+  })
+
+  it("should render event details correctly", () => {
+    cy.get(".postings-container")
+      .should("contain", "Color Run")
```
Comment (coopterrones) [inline cypress/integration/EventView_test.js:12]: and and and and and

Comment (coopterrones) [reply cypress/integration/EventView_test.js:12]: Im curious is we can consolidate somehow here but I get the idea

**Review by okalimbet — APPROVED** [review]
Nice changes, we do need to figure out cypress and probably ask for help

File: cypress/integration/EventView_test.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        "rgb(46, 196, 182)"
+        )
+  });
+
+  it("Return to home page button should display and function", () => {
+    cy.get(".back-button-img")
+      .should("be.visible")
+      .click();
+
+    cy.url().should("include", "/")
```
Comment (coopterrones) [inline cypress/integration/EventView_test.js:42]: I love how concise cypress is when testing navigation through the site

File: cypress/integration/EventView_test.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+    cy.get(".posting-position-cards-wrapper")
+      .should("contain", "assisting with check-in")
+      .and("contain", "Open Spots: 3")
+      .and("contain", "tossing color")
+      .and("contain", "Open Spots: 5")
+      .and("contain", "handing out water")
+      .and("contain", "Open Spots: 6");
+
+    cy.get(".posting-positions-card:first")
```
Comment (coopterrones) [inline cypress/integration/EventView_test.js:63]: nice

**Pull request merged**